### DISCUSSION TITLE
random forest classifier (Attribute error)

### DISCUSSION THREAD

**Discussion by SyedTuhinAli** [semiautomaticgit/SemiAutomaticClassificationPlugin#155] [Q&A] [answered]
I am using QGIS Semi-automatic Plugins. When I try to run the Random Forest classifier, the following error displayed "AttributeError: 'str' object has no attribute 'setContrastEnhancement' " please find the error text below:

[random forest error.txt](https://github.com/semiautomaticgit/SemiAutomaticClassificationPlugin/files/5974031/random.forest.error.txt)

Qgis: v 3.16.3
Semi-automatic Plugins: v. 7.6.0
Python: built-in with the Qgis

snap gpt executable file "C:\Program Files\snap\bin\gpt.exe" in the processing setting has already provided

The maximum-likelihood classifier works.

Thanks

**Comment by semiautomaticgit** [comment, marked as answer]
Hello,

could you please provide information about Windows version and how QGIS was installed?

Thank you

**Reply by SyedTuhinAli** [reply]
Thanks for your reply 

I’m working on windows 7 64 bit

I had already installed qgis 2.7 before installing latest version because I wanted to use a plugin which is unavailable in the new version. 

Moreover, python 3.6 was already installed

**Reply by semiautomaticgit** [reply]
You are welcome.
Did you install QGIS through the standalone package or using OSGEO installer?

**Reply by SyedTuhinAli** [reply]
I installed it through the standalone package

**Reply by SyedTuhinAli** [reply]
![Capture2](https://user-images.githubusercontent.com/78980669/107823522-3a857b80-6d80-11eb-9ed6-6104eb996744.JPG)

This is the one I used

**Reply by semiautomaticgit** [reply]
Thank you, I'll look into this

**Reply by semiautomaticgit** [reply]
Please update the plugin to version 7.6.1, it should work now

**Reply by SyedTuhinAli** [reply]
Thanks , it worked

**Reply by semiautomaticgit** [reply]
You are very welcome

**Reply by SyedTuhinAli** [reply]
Dear Luca, 

I am now facing a new problem with the random forest classifier. It doesn't work for macro class id. Although it works for class id under macro (c_id), it gives the wrong output. Please find my classification snapshot (jpeg), Python error report for macro class (error), and the wrong output/results (output) for the c_id classification.

Thanks 

[output for C ID.txt](https://github.com/semiautomaticgit/SemiAutomaticClassificationPlugin/files/5983016/output.for.C.ID.txt)
[error.txt](https://github.com/semiautomaticgit/SemiAutomaticClassificationPlugin/files/5983017/error.txt)
![train](https://user-images.githubusercontent.com/78980669/107975124-e82e9f80-6fb7-11eb-9a79-5009be05c0b5.JPG)

**Reply by semiautomaticgit** [reply]
Please update the plugin to version 7.6.3, it should work.
Please note that C_IDs should be different for every ROI, even if belonging to different MC_IDs

**Reply by SyedTuhinAli** [reply]
Thanks! It is working now.
